src/VisualStudio/Core/Def/Utilities/ClipboardHelpers.cs

```diff
@@ -0,0 +1,57 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Runtime.InteropServices;
+using System.Text;
+using System.Threading.Tasks;
+using System.Windows;
+
+namespace Microsoft.VisualStudio.LanguageServices.Utilities
+{
+    internal static class ClipboardHelpers
+    {
+        [DllImport("user32.dll")]
+        [PreserveSig]
+        private static extern int GetPriorityClipboardFormat([In] uint[] formatPriorityList, [In] int formatCount);
+
+        /// <summary>
+        /// Determines if clipboard data exists for the text format without blocking
+        /// on <see cref="Clipboard.GetText()"/> or <see cref="Clipboard.GetDataObject()"/> for cases where the clipboard is being
+        /// delay rendered. See https://devdiv.visualstudio.com/DevDiv/_workitems/edit/1463413/ 
+        /// </summary>
+        public static bool CanGetText()
+        {
+            const uint TextFormatId = 1;          // ID for clipboard text format (CF_TEXT)
+
+            var formats = new uint[] { TextFormatId };
+
+            // GetPriorityClipboardFormat returns 0 if the clipboard is empty, -1 if there is data in the clipboard but it does not match
+            // any of the input formats and otherwise returns the ID of the format available in the clipboard.
+            return GetPriorityClipboardFormat(formats, formats.Length) > 0;
+        }
+
+        /// <summary>
+        /// Gets the text from the clipboard. Check <see cref="CanGetText"/>
+        /// first before calling this to verify text can be safely retrieved
+        /// without a deadlock
```

Comment:
This comment is misleading. CanGetText definitely does NOT check if one can retrieve the data without a deadlock, it checks if the client side (cached) clipboard fmt list contains CF_TEXT. This at best avoids an RPC call to the clipboard content owner process (if not the current process), but it certainly doesn't guarantee any such call will succeed. I am not sure if Windows eagerly caches CF_TEXT contents client side, but if it does it implies the client has made at least one clipboard API call since the clipboard contents changed last, and that call itself could deadlock. End of the day anytime anyone makes a cross-proc RPC call it can deadlock.